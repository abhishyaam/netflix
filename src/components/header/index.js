import React from 'react';
import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSeacrhActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSeacrhActive((searchActive) => !searchActive)}
      >
        <img src='/images/icons/search.png' alt='Search' />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder='Search films and series'
        active={searchActive}
      />
    </Search>
  );
};
