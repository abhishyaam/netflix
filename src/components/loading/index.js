import React from 'react';
import { Picture, Spinner, LockBody, ReleaseBody } from './styles/loading';

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid='loading-picture' />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingRealeaseBody({ src, ...restProps }) {
  return <ReleaseBody />;
};
