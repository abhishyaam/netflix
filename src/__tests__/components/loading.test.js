import React from 'react';
import { render } from '@testing-library/react';

import { Loading } from '../../components';

describe('<Loading />', () => {
  it('renders <Loading/>', () => {
    const { container, getByTestId } = render(
      <Loading src='/images/abhi.jpg' data-testid='loading' />
    );
    expect(getByTestId('loading')).toBeTruthy();
    expect(getByTestId('loading-picture')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders <Loading.ReleaseBody/>', () => {
    const { container, getByTestId } = render(
      <Loading.ReleaseBody data-testid='loading' />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
