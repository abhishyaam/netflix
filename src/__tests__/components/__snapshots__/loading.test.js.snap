// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders <Loading.ReleaseBody/> 1`] = `null`;

exports[`<Loading /> renders <Loading/> 1`] = `
<div
  class="sc-jRQBWg fUKguS"
  data-testid="loading"
>
  <img
    class="sc-gKclnd fgPqYR"
    data-testid="loading-picture"
    src="/images/users//images/abhi.jpg.png"
  />
</div>
`;
